interpreter/llm/setup_llm.py

```diff
@@ -5,13 +5,48 @@
 from .setup_openai_coding_llm import setup_openai_coding_llm
 import os
 import litellm
+import sys
+import platform
+import subprocess
+
+
 
 def setup_llm(interpreter):
     """
     Takes an Interpreter (which includes a ton of LLM settings),
     returns a Coding LLM (a generator that streams deltas with `message` and `code`).
     """
 
+    # Get some useful system information before starting the LLM
+    if interpreter.debug_mode:
+        
+        # Get Python version
+        python_version = sys.version
+        print(f"\nPython Version: {python_version}")
+
+        # Get pip version
+        try:
+            pip_version = subprocess.getoutput('pip --version')
+            print(f"Pip Version: {pip_version}")
+        except Exception as e:
+            print(f"Error retrieving pip version: {e}")
+
+        # Get platform (i.e., Windows, Linux, etc.s)
+        platform_name = sys.platform
+        print(f"Platform: {platform_name}")
+
+        # More detailed OS info using the platform module
+        os_name = platform.system()
+        os_release = platform.release()
+        os_version = platform.version()
+        print(f"OS Name: {os_name}")
+        print(f"OS Release: {os_release}")
+        print(f"OS Version: {os_version}")
+
+        # Print the architecture as well:
+        architecture = platform.architecture()
+        print(f"Architecture: {architecture[0]}")
```

Comment:
I think it would be awesome to move all of this into [`interpreter/utils`](https://github.com/KillianLucas/open-interpreter/tree/main/interpreter/utils) as a `system_info` utility, or something like that, and then add a condition to the [`handle_debug()` magic command handler for `%debug`](https://github.com/KillianLucas/open-interpreter/blob/main/interpreter/terminal_interface/magic_commands.py#L66) to accept an `info` or `system` argument (not sure what the best name is) and then call this new utility so that folks can dynamically generate this debug info to copy and paste into Issues.